### PR TITLE
Add default package visibility to origins

### PR DESCRIPTION
**Pull request by raskchanky**
This adds a setting called `default_package_visibility` to origins.  It defaults to `public` for existing origins, but it's a required field when you create an origin via the API.

Note that this PR doesn't do anything regarding the showing or hiding of packages based on this setting.  This just provides the setting.

Also note that since origins have up to this point just been a name, there is currently no API to update an origin, so once this setting is set, you can't change it.  If we anticipate needing the ability to toggle this setting, we'll need to make an API to do so.

tenor-195429647

Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-originsrv/src/migrations/origins.rs [hunk 1]
```diff
@@ -117,5 +117,26 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
                         UPDATE origins SET session_sync = true WHERE id = in_origin_id;
                     END
                     $$ LANGUAGE plpgsql VOLATILE"#)?;
+    migrator.migrate("originsrv",
```
Comment (mwrock) [inline components/builder-originsrv/src/migrations/origins.rs]: pretty sure the key needs to be incremented to something like `originsrv-v7`

Comment (raskchanky) [reply components/builder-originsrv/src/migrations/origins.rs]: Nah, that hasn't been required since the migrator was rewritten months ago.

**Comment by mwrock** [comment]
Looks like the data store tests are failing. Also might be good to add an assert for this field in the create origin test.

File: components/builder-originsrv/src/migrations/origins.rs (hunk 1 above)
Comment (mwrock) [reply components/builder-originsrv/src/migrations/origins.rs]: oh ok!

**Comment by raskchanky** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/272647242).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/272647242).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**